session106/XMap/XMap/Model/TFLFetcher.swift

```diff
@@ -33,10 +33,16 @@ class TFLFetcher: NSObject {
         executor.execute { (responseJSON) in
             
             // Received Data for parsing
-            // Do some work...Mahyad?
+            
+            var parsedStations = [Station]()
+                
+            if let stationData = try? JSONSerialization.data(withJSONObject: responseJSON, options: []), let nearbyStation = try? JSONDecoder().decode(Station.self, from: stationData) {
+                parsedStations.append(nearbyStation)
+                print(nearbyStation)
+            }
             
             DispatchQueue.main.async {
-                handler?(responseJSON)
+                handler?(parsedStations)
```

Comment:
The handler: `TFLFetcherStationsHandler` should not be `(Any) -> ()`, it should be `([Station]) -> ()`